### PR TITLE
Patch to avoid lookup tables

### PR DESCRIPTION
**Pull request by Pankul94**
Lookup tables can lead to cache-timing leak risk and this change is to avoid lookup tables.

I have ported this from https://github.com/hashicorp/vault/commit/ed08e45069d60fc4452fad7d44bcfd3b2e4c8236 and verified the test cases pass.

### CODE REVIEW

**Comment by konidev20** [comment]
Hey @Pankul94, 

Thank you for reviewing our project and pushing this fix. 

I will spare some time for this on the weekend. 

Regards, 
@konidev20

**Comment by konidev20** [comment]
Hey @Pankul94, 

I believe the pipeline is trying to run on your account. 

Since you don't have the Sonar Cloud Token to execute the Action it is failing. 

I will check and fix this too.

**Comment by konidev20** [comment]
Hey @Pankul94, 

It looks like SonarCloud does not support scanning external PRs (PRs raised from forks or non-collaborators). Ref: https://community.sonarsource.com/t/sonar-token-permissions-in-pull-request-github-actions/90614/7

I will see how we can run the security scans for this.

File: pyshamir/_utils.py
Comment (konidev20) [inline pyshamir/_utils.py:45]: I guess we can remove this comment since we are not using division in the Galois field using lookup tables anymore.

**Review by konidev20 — APPROVED** [review]
Everything looks good to me. I will wait for @Loadingname91  and he can merge. Once it's merged the SonarScan will run on the main branch, and if everything is good, I will trigger a new release on PyPi.

File: pyshamir/_utils.py
Comment (Pankul94) [reply pyshamir/_utils.py]: Removed

**Comment by Pankul94** [comment]
Sounds good

**Review by Loadingname91 — APPROVED** [review]
Seems good to me

**Comment by konidev20** [comment]
Hey @Pankul94, 

The changes have been merged and the SonarCloud checks have passed. 

Thank you for your contributions :)

**Pull request merged**
